src/zjs_dgram.c

```diff
@@ -0,0 +1,334 @@
+    }
+    int offset = (int)jerry_get_number_value(argv[1]);
+    int len = (int)jerry_get_number_value(argv[2]);
+    if (offset + len > buf->bufsize) {
+        return zjs_error("offset/len beyond buffer end");
+    }
+
+    sa_family_t family = net_context_get_family(handle->udp_sock);
+    struct sockaddr sockaddr_buf;
+    get_addr(family, argv[4], argv[3], &sockaddr_buf);
+
+    struct net_buf *send_buf = net_nbuf_get_tx(handle->udp_sock);
+    if (!net_nbuf_append(send_buf, len, buf->buffer + offset)) {
+        return zjs_error("netbuf error");
+    }
+
+    void *user_data = NULL;
+    if (argc > 5) {
+        user_data = (void*)argv[5];
+    }
+
+    CHECK(net_context_sendto(send_buf, &sockaddr_buf, sizeof(sockaddr_buf),
+                             udp_sent, K_NO_WAIT, NULL, user_data));
+
+    return ZJS_UNDEFINED;
+}
+
+static jerry_value_t zjs_dgram_sock_bind(const jerry_value_t function_obj,
+                                         const jerry_value_t this,
+                                         const jerry_value_t argv[],
+                                         const jerry_length_t argc)
+{
+    int ret;
+
+    if (argc != 2)
+        return zjs_error("dgram.bind: wrong number of args");
+
+    GET_HANDLE(dgram_handle_t, handle);
+
+    sa_family_t family = net_context_get_family(handle->udp_sock);
+    struct sockaddr sockaddr_buf;
+    get_addr(family, argv[1], argv[0], &sockaddr_buf);
+
+    CHECK(net_context_bind(handle->udp_sock, &sockaddr_buf,
+                               sizeof(sockaddr_buf)));
+    // See comment in createSocket() why this is called here
+    CHECK(net_context_recv(handle->udp_sock, udp_received, K_NO_WAIT, handle));
+
+    return ZJS_UNDEFINED;
+}
+
+static jerry_value_t zjs_dgram_sock_close(const jerry_value_t function_obj,
+                                          const jerry_value_t this,
+                                          const jerry_value_t argv[],
+                                          const jerry_length_t argc)
+{
+    GET_HANDLE(dgram_handle_t, handle);
+    zjs_dgram_free_cb((uintptr_t)handle);
+    return ZJS_UNDEFINED;
+}
+
+jerry_value_t zjs_dgram_init()
+{
+    // TODO: Interface address initialization doesn't belong to this module
+    static struct in_addr in4addr_my = { { { 192, 0, 2, 1 } } };
+    // 2001:db8::1
+    static struct in6_addr in6addr_my = { { { 0x20, 0x01, 0x0d, 0xb8, 0, 0, 0, 0,
+                                              0, 0, 0, 0, 0, 0, 0, 0x1 } } };
+    net_if_ipv4_addr_add(net_if_get_default(), &in4addr_my, NET_ADDR_MANUAL, 0);
+    net_if_ipv6_addr_add(net_if_get_default(), &in6addr_my, NET_ADDR_MANUAL, 0);
+
+    // create socket prototype object
+    zjs_native_func_t array[] = {
+        { zjs_dgram_sock_on, "on" },
+        { zjs_dgram_sock_send, "send" },
+        { zjs_dgram_sock_bind, "bind" },
+        { zjs_dgram_sock_close, "close" },
+        { NULL, NULL }
+    };
+    zjs_dgram_socket_prototype = jerry_create_object();
```

Comment:
Will do if you guys think it's important thing to do for embedded implementation.